rpcserver.go

```diff
@@ -6946,6 +6946,134 @@ func (r *rpcServer) SendPayment(req *tchrpc.SendPaymentRequest,
+
+	if req.InvoiceRequest == nil {
+		return nil, fmt.Errorf("invoice request must be specified")
+	}
+	iReq := req.InvoiceRequest
+
+	// Do some preliminary checks on the asset ID and make sure we have any
+	// balance for that asset.
+	if len(req.AssetId) != sha256.Size {
+		return nil, fmt.Errorf("asset ID must be 32 bytes")
+	}
+	var assetID asset.ID
+	copy(assetID[:], req.AssetId)
+
+	// The peer public key is optional if there is only a single asset
+	// channel.
+	var peerPubKey *route.Vertex
+	if len(req.PeerPubkey) > 0 {
+		parsedKey, err := route.NewVertexFromBytes(req.PeerPubkey)
+		if err != nil {
+			return nil, fmt.Errorf("error parsing peer pubkey: %w",
+				err)
+		}
+
+		peerPubKey = &parsedKey
+	}
+
+	// We can now query the asset channels we have.
+	assetChan, err := r.rfqChannel(ctx, assetID, peerPubKey)
+	if err != nil {
+		return nil, fmt.Errorf("error finding asset channel to use: %w",
+			err)
+	}
+
+	// Even if the user didn't specify the peer public key before, we
+	// definitely know it now. So let's make sure it's always set.
+	peerPubKey = &assetChan.channelInfo.PubKeyBytes
+
+	expirySeconds := iReq.Expiry
+	if expirySeconds == 0 {
+		expirySeconds = int64(rfq.DefaultInvoiceExpiry.Seconds())
+	}
+	expiryTimestamp := time.Now().Add(
+		time.Duration(expirySeconds) * time.Second,
+	)
+
+	resp, err := r.AddAssetBuyOrder(ctx, &rfqrpc.AddAssetBuyOrderRequest{
+		AssetSpecifier: &rfqrpc.AssetSpecifier{
+			Id: &rfqrpc.AssetSpecifier_AssetId{
+				AssetId: assetID[:],
+			},
+		},
+		MinAssetAmount: req.AssetAmount,
+		Expiry:         uint64(expiryTimestamp.Unix()),
+		PeerPubKey:     peerPubKey[:],
+		TimeoutSeconds: uint32(
+			rfq.DefaultTimeout.Seconds(),
+		),
+	})
+	if err != nil {
+		return nil, fmt.Errorf("error adding buy order: %w", err)
+	}
+
+	var acceptedQuote *rfqrpc.PeerAcceptedBuyQuote
+	switch r := resp.Response.(type) {
+	case *rfqrpc.AddAssetBuyOrderResponse_AcceptedQuote:
+		acceptedQuote = r.AcceptedQuote
+
+	case *rfqrpc.AddAssetBuyOrderResponse_InvalidQuote:
+		return nil, fmt.Errorf("peer %v sent back an invalid quote, "+
+			"status: %v", r.InvalidQuote.Peer,
+			r.InvalidQuote.Status.String())
+
+	case *rfqrpc.AddAssetBuyOrderResponse_RejectedQuote:
+		return nil, fmt.Errorf("peer %v rejected the quote, code: %v, "+
+			"error message: %v", r.RejectedQuote.Peer,
+			r.RejectedQuote.ErrorCode, r.RejectedQuote.ErrorMessage)
+
+	default:
+		return nil, fmt.Errorf("unexpected response type: %T", r)
+	}
+
+	// Now that we have the accepted quote, we know the amount in Satoshi
+	// that we need to pay. We can now update the invoice with this amount.
+	mSatPerUnit := acceptedQuote.AskPrice
+	iReq.ValueMsat = int64(req.AssetAmount * mSatPerUnit)
+
+	// The last step is to create a hop hint that includes the fake SCID of
+	// the quote, alongside the channel's routing policy.
+	channelID := assetChan.channelInfo.ChannelID
+	ourPolicy, err := r.getOurPolicy(ctx, channelID, peerPubKey.String())
```

Comment:
Perhaps instead of `ourPolicy` in both variable name and method name, just `channelPolicy`? Because "their" policy isn't something we deal with. I have the same sort of thoughts around the `getOurPolicy` method name and doc.